Scripts/buildStaticLibraries.sh

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+#
+# Usage: ./Scripts/buildStaticLibraries.sh
+#
+# This script creates static libraries for our libraries bundled in xcframeworks. 
+# It also creates .bundle files that need to be copied for the libraries to work
+
+products_path="Products"
+derived_data="DerivedData"
+
+archives="$products_path/Archives"
+framework_path="$products_path/Library/Frameworks"
+
+platforms=(
+  "simulator"
+  "os"
+)
+
+artifacts=(
+  "StreamChatUI"
+  "StreamChat"
+)
+
+library="StreamChatUI"
+library_target=$library"Static"
+
+mkdir -p DerivedData
+
+for platform in ${platforms[@]}
+do
+  destination=""
+  if [[ "$platform" == "os" ]]
+  then
+    destination="generic/platform=iOS"
+  else
+    destination="generic/platform=iOS Simulator"
+  fi
+
+  logs_file="$derived_data/build_static_libraries_iphone$platform-xcodebuild.log"
+  echo "→ Creating archive for iphone$platform. Destination: $destination."
+  echo "→ Building... Logs available: $logs_file"
+
+  # We only archive StreamChatUI because since it has a dependency on StreamChat, it will build this one too.
+  xcodebuild archive \
+          -project StreamChat.xcodeproj \
+          -scheme "$library_target" \
+          -destination "$destination" \
+          -derivedDataPath $derived_data \
+          -archivePath "$archives/$library-iphone-$platform" \
+          SKIP_INSTALL=NO \
+          BUILD_LIBRARY_FOR_DISTRIBUTION=YES > $logs_file
+
+  # Unlike `xcodebuild build`, `xcodebuild archive` does not generate a .swiftmodule for static libraries. So we manually
+  # copy it from our local DerivedData
+  #
+  # Expanded explanation:
+  # We cannot use `xcodebuild build` artifacts because those are build with the flag -fembed-bitcode-marker.
```

Comment:
Hey @polqf,
wouldn't adding the appropriate env var overrides to `xcodebuild build` resolve the issue around the bitcode?

eg
```
xcodebuild build \
....
....
OTHER_CFLAGS="-fembed-bitcode" \ 
BITCODE_GENERATION_MODE="bitcode" \ 
ENABLE_BITCODE=YES
```

?

Reply:
I tried, but passing -fembed-bitcode in OTHER_CFLAGS does not take precedence over the already defined -fembed-bitcode-marker in xcodebuild 🤷‍♂️ .